core/src/main/java/io/grpc/internal/DnsNameResolver.java

```diff
@@ -200,17 +217,33 @@ public void run() {
           }
 
           ResolutionResults resolutionResults;
-          try {
-            ResourceResolver resourceResolver = null;
-            if (enableJndi) {
-              resourceResolver = getResourceResolver();
+          long networkAddressCacheTtl = getNetworkAddressCacheTtl();
```

Comment:
Let's just do this once for the class. We're not concerned about the value changing over time (because Java won't respect the change anyway).